Source/BluetoothManager.swift

```diff
@@ -342,22 +342,29 @@ public class BluetoothManager {
     /// - Parameter peripheral: `Peripheral` which is monitored for connection.
     /// - Returns: Observable which emits next events when `peripheral` was connected.
     public func monitorConnection(for peripheral: Peripheral) -> Observable<Peripheral> {
-        return monitorPeripheral(on: delegateWrapper.rx_didConnectPeripheral, peripheral: peripheral)
+        return monitorPeripheral(
+            on: delegateWrapper.rx_didConnectPeripheral.map { ($0, nil) },
+            peripheral: peripheral
+        ).map { $0.0 }
     }
 
     /// Emits `Peripheral` instance when it's disconnected.
     /// - Parameter peripheral: `Peripheral` which is monitored for disconnection.
-    /// - Returns: Observable which emits next events when `peripheral` was disconnected.
-    public func monitorDisconnection(for peripheral: Peripheral) -> Observable<Peripheral> {
-        return monitorPeripheral(on: delegateWrapper.rx_didDisconnectPeripheral.map { $0.0 }, peripheral: peripheral)
+    /// - Returns: Observable which emits next events when `peripheral` was disconnected and error which is
+    /// a disconnection reason when it was caused by 'peripheral' otherwise it's nil.
+    public typealias DisconnectionReason = Error?
```

Comment:
type alias should be above the documentation comments. Right now docs refer to type alias not to a function. Additionaly it should be `DisconnectionReason = Error`. Then in interface user sees immediately that it's optional — without looking at type alias definition.